### PR TITLE
Add KubernetesProbesOptionsValidator to ensure different ports for probes

### PR DESCRIPTION
**Pull request by makazeu**
Add the KubernetesProbesOptionsValidator to check if the 3 Kubernetes probes (liveness, startup, readiness) use different ports.

### Context
If we set same ports for Kubernetes probes, it will cause issue below:
```
        builder.Services.AddKubernetesProbes(options =>
        {
            options.LivenessProbe.TcpPort = 2305;
            options.StartupProbe.TcpPort = 2305;
            options.ReadinessProbe.TcpPort = 2305;
        });
```

```
fail: Microsoft.Extensions.Diagnostics.Probes.TcpEndpointProbesService[1466257416]
      Error updating health status through TCP endpoint
      System.Net.Sockets.SocketException (10048): Only one usage of each socket address (protocol/network address/port) is normally permitted.
         at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, Boolean disconnectOnFailure, String callerName)
         at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)
         at System.Net.Sockets.Socket.Bind(EndPoint localEP)
         at System.Net.Sockets.TcpListener.Start(Int32 backlog)
         at Microsoft.Extensions.Diagnostics.Probes.TcpEndpointProbesService.UpdateHealthStatusAsync(CancellationToken cancellationToken)
```
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/extensions/pull/5400)

### CODE REVIEW

**Comment by RussKie** [comment]
@mobratil please triage and review

File: test/Libraries/Microsoft.Extensions.Diagnostics.Probes.Tests/KubernetesProbesOptionsValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace Microsoft.Extensions.Diagnostics.Probes.Test;
+
+public class KubernetesProbesOptionsValidatorTests
```
Comment (mobratil) [inline test/Libraries/Microsoft.Extensions.Diagnostics.Probes.Tests/KubernetesProbesOptionsValidatorTests.cs:8]: Can we also add another test that KubernetesProbesOptions are validated when we use AddKubernetesProbes method?

Comment (makazeu) [reply test/Libraries/Microsoft.Extensions.Diagnostics.Probes.Tests/KubernetesProbesOptionsValidatorTests.cs:13]: Hi @mobratil, I've updated the unit tests, please check `Validator_WhenHostStarts_Fails` and `Validator_WhenHostStarts_Succeeds`? Thank you!

**Review by mobratil — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Pull request merged**